Sources/UInt128/UInt128.swift

```diff
@@ -73,14 +73,15 @@ public struct UInt128 {
     }
 
     /// Initialize a UInt128 value from a string.
+    /// Returns `nil` if input cannot be converted to a UInt128 value.
     ///
     /// - parameter source: the string that will be converted into a
     ///   UInt128 value. Defaults to being analyzed as a base10 number,
     ///   but can be prefixed with `0b` for base2, `0o` for base8
     ///   or `0x` for base16.
-    public init(_ source: String) throws {
+    public init?(_ source: String) {
         guard let result = UInt128._valueFromString(source) else {
-            throw UInt128Errors.invalidString
+            return nil
         }
         self = result
     }
```

Comment:
I was wondering why you feel its necessary to change this from a throwable to an optional. Would you mind elaborating on why you want to break this existing API contract?

Reply:
Addressed in https://github.com/Jitsusama/UInt128/pull/37#issuecomment-1032113218.